### DISCUSSION TITLE
How can I fill out url field with display_url in http redirect task of workflow,please?

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#15022] [Q&A]
Hi,
How can I fill out url field with display_url in http redirect task of workflow,please?

![image](https://github.com/OrchardCMS/OrchardCore/assets/51013754/a630a339-e7dc-4dc6-adc4-a9998f5af800)

{{ Workflow.Input.ContentItem | display_url }} is not correct.

**Comment by MikeAlhayek** [comment]
Maybe the ContentItem is not in the input 

try

```
{{ Workflow.ContentItem | display_url }}
```

If you are trying to redirect back to a form location. In OC 1.8 we have a new task that would allow you to redirect the user back to where the form was originated from.

**Comment by effapp** [comment]
Hi MikeAlayek,
       Can you add the property display_url to ContentItem like display_text ?

**Comment by effapp** [comment]
> Maybe the ContentItem is not in the input
> 
> try
> 
> ```
> {{ Workflow.ContentItem | display_url }}
> ```
> 
> If you are trying to redirect back to a form location. In OC 1.8 we have a new task that would allow you to redirect the user back to where the form was originated from.

Hi, 

{{ Workflow.ContentItem | display_url }} throw exception, Workflow.ContentItem is null

**Comment by MikeAlhayek** [comment]
Then `{{ Workflow.Input.ContentItem | display_url }}` should have worked assuming your workflow event is a content item based.

What event are you using to start your workflow?

**Comment by effapp** [comment]
> Then `{{ Workflow.Input.ContentItem | display_url }}` should have worked assuming your workflow event is a content item based.
> 
> What event are you using to start your workflow?

I used Http Request Event to start my workflow.  {{ Workflow.Input.ContentItem | display_url }} render nothing.

**Comment by MikeAlhayek** [comment]
I think that is what you are doing wrong. I think for ContentItem to exists, you'll need to add a content item event like Created, updated or published content

**Comment by effapp** [comment]
> I think that is what you are doing wrong. I think for ContentItem to exists, you'll need to add a content item event like Created, updated or published content

If I want to use Http Request Event to start my workflow and want to  ContentItem to exists, What should I do,please?

**Comment by MikeAlhayek** [comment]
What is the logic that you are trying to handle in workflow?

**Comment by effapp** [comment]
> What is the logic that you are trying to handle in workflow?

I'll submit the form of the ContentItem display page to trigger workflow, then it redirect to  {{ Workflow.Input.ContentItem | display_url }}  in the last activity "http redirect task" of workflow.

**Comment by zl2fxy** [comment]
Your workflow doesn't seem to be initiated by content items, so there's no input inside. Can I take a look at your workflow page? I need to know more information to assist you.

**Comment by effapp** [comment]
> Your workflow doesn't seem to be initiated by content items, so there's no input inside. Can I take a look at your workflow page? I need to know more information to assist you.

Hi, the answer is shown as following figure:
![image](https://github.com/OrchardCMS/OrchardCore/assets/51013754/d0de0685-9f87-4417-94e8-e8c0010bef5a)
